docker/web/Dockerfile

```diff
@@ -0,0 +1,125 @@
+###############################################################################
+#-----------------------------    BUILD STAGE   ------------------------------#
+###############################################################################
+
+FROM ubuntu:bionic as builder
+
+ARG CC_VERSION=master
+ENV CC_VERSION ${CC_VERSION}
+
+ARG CC_DATABASE=sqlite
+ENV CC_DATABASE ${CC_DATABASE}
+
+ARG CC_BUILD_TYPE=Release
+ENV CC_BUILD_TYPE ${CC_BUILD_TYPE}
+
+RUN set -x && apt-get update -qq \
+  && apt-get install -qqy \
+    cmake make \
+    llvm-7 clang-7 llvm-7-dev libclang-7-dev \
+    libboost-all-dev \
+    odb libodb-sqlite-dev libodb-pgsql-dev \
+    libsqlite3-dev \
+    default-jdk ant\
+    libssl1.0-dev \
+    libgraphviz-dev \
+    libmagic-dev \
+    libgit2-dev \
+    nodejs-dev node-gyp npm \
+    ctags \
+    wget \
+    git \
+    libgtest-dev \
+    && \
+    # Build thrift.
+    wget "http://www.apache.org/dyn/mirrors/mirrors.cgi?action=download&filename=thrift/0.12.0/thrift-0.12.0.tar.gz" -O /opt/thrift-0.12.0.tar.gz; \
+    tar -xf /opt/thrift-0.12.0.tar.gz -C /opt; \
+    cd /opt/thrift-0.12.0; \
+    ./configure --prefix=/opt/thrift --silent; \
+    cd /; \
+    make -C /opt/thrift-0.12.0 -j4 --silent install; \
+    rm -rf /opt/thrift-0.12.0 /opt/thrift-0.12.0.tar.gz; \
+    # Build GTest.
+    cd /usr/src/googletest/ && \
+    mkdir build && \
+    cd build && \
+    cmake .. && \
+    make install && \
+    cd / && \
+    rm -rf /usr/src/googletest/build
+
+ENV PATH="/opt/thrift/bin:$PATH"
+ENV CMAKE_PREFIX_PATH="/opt/thrift:$CMAKE_PREFIX_PATH"
+
+# Download CodeCompass release.
+RUN git clone https://github.com/Ericsson/CodeCompass.git /CodeCompass
+WORKDIR /CodeCompass
+RUN git checkout ${CC_VERSION}
+
+# Build CodeCompass.
+RUN mkdir /CodeCompass-build && \
+  cd /CodeCompass-build && \
+  cmake /CodeCompass \
+    -DDATABASE=$CC_DATABASE \
+    -DCMAKE_INSTALL_PREFIX=/CodeCompass-install \
+    -DCMAKE_BUILD_TYPE=$CC_BUILD_TYPE && \
+  make && \
+  make install
+
+###############################################################################
+#--------------------------    PRODUCTION STAGE   ----------------------------#
+###############################################################################
+FROM ubuntu:bionic
+
+RUN set -x && apt-get update -qq \
+  && apt-get install -qqy --no-install-recommends \
+    llvm-7 \
+    libboost-filesystem-dev libboost-log-dev libboost-program-options-dev \
+    odb libodb-sqlite-dev libodb-pgsql-dev \
+    libsqlite3-dev \
+    default-jre \
+    libgit2-dev \
+    libssl1.0.0 \
+    libgvc6 \
+    # To switch user and exec command.
+    gosu \
+  && apt-get clean \
+  && rm -rf /var/lib/apt/lists/ \
+  && set +x
+
+# Copy CodeCompass installed directory.
+COPY --from=builder /CodeCompass-install /codecompass
```

Comment:
COPY is able to handle permissions changes changes with the ` --chown` flag so there is no need to make a RUN for that.